templates/components/agents/typescript/financial_report/app/api/chat/workflow/factory.ts

```diff
@@ -0,0 +1,30 @@
+import { ChatMessage, ToolCallLLM } from "llamaindex";
+import { FinancialReportWorkflow } from "./finReport";
+import { getAvailableTools } from "./tools";
+
+const TIMEOUT = 360 * 1000;
+
+export async function createWorkflow(options: {
+  chatHistory: ChatMessage[];
+  llm?: ToolCallLLM;
+}) {
+  const tools = await getAvailableTools();
+  const queryEngineTools = tools.filter((tool) =>
```

Comment:
Wasnt this shorter before?

Reply:
Now it reuses the shared code, so it should have multiple query engine tools if the index is LlamaCloud